bool open
float32 angle
